components/core/src/clp_s/ArchiveWriter.cpp

```diff
@@ -138,19 +142,89 @@ size_t ArchiveWriter::store_tables() {
             FileWriter::OpenMode::CreateForWriting
     );
     m_table_metadata_compressor.open(m_table_metadata_file_writer, m_compression_level);
-    m_table_metadata_compressor.write_numeric_value(m_id_to_schema_writer.size());
-    for (auto& i : m_id_to_schema_writer) {
-        m_table_metadata_compressor.write_numeric_value(i.first);
-        m_table_metadata_compressor.write_numeric_value(i.second->get_num_messages());
-        m_table_metadata_compressor.write_numeric_value(m_tables_file_writer.get_pos());
 
-        m_tables_compressor.open(m_tables_file_writer, m_compression_level);
-        size_t uncompressed_size = i.second->store(m_tables_compressor);
-        m_tables_compressor.close();
-        delete i.second;
+    /**
+     * Packed stream metadata schema
+     * # num packed streams <64 bit>
+     * # [offset into file <64 bit> uncompressed size <64 bit>]+
+     * # num_separate_column_schemas (always 0 in current implementation)
+     * # [undefined]+ (will be defined once supported)
+     * # num schemas <64 bit>
+     * # [stream id <64 bit> offset into stream <64 bit> schema id <32 bit> num messages <64 bit>]+
```

Comment:
Can we make it more clear to something like 
```suggestion
* This metadata schema organizes information about packed compression streams and associated 
 * schema tables within a file. The schema is divided into two primary sections:
 * Section 1: Compression Streams Metadata
 * - Contains metadata about each compression stream.
 * - Structure:
 *   - Number of packed streams: <64-bit integer>
 *   - For each stream: 
 *     - Offset into the file: <64-bit integer>
 *     - Uncompressed size: <64-bit integer>
 * - `num_separate_column_schemas` is always 0 in the current implementation.
 * - Undefined section for separate column schemas, reserved for future support.
 * 
 * Section 2: Schema Tables Metadata
 * - Contains metadata about schema tables associated with each compression stream.
 * - Structure:
 *   - Number of schema tables: <64-bit integer>
 *   - For each schema table:
 *     - Stream ID: <64-bit integer>
 *     - Offset into the stream: <64-bit integer>
 *     - Schema ID: <32-bit integer>
 *     - Number of messages: <64-bit integer>
```
and probably move some texts starting from line 155 here